python/hail/methods/statgen.py

```diff
@@ -104,6 +103,140 @@ def ibd(dataset, maf=None, bounded=True, min=None, max=None):
                                               joption(max)))
 
 @handle_py4j
+@typecheck(call=expr_call,
+           aaf_threshold=numeric,
+           include_par=bool,
+           female_threshold=numeric,
+           male_threshold=numeric,
+           aaf=nullable(strlike))
+def impute_sex(call, aaf_threshold=0.0, include_par=False, female_threshold=0.2, male_threshold=0.8, aaf=None):
+    """Impute sex of samples by calculating inbreeding coefficient on the X
+    chromosome.
+
+    .. include:: ../_templates/req_tvariant.rst
+
+    .. include:: ../_templates/req_biallelic.rst
+
+    Examples
+    --------
+
+    Remove samples where imputed sex does not equal reported sex:
+
+    >>> imputed_sex = methods.impute_sex(ds.GT)
+    >>> ds.filter_cols(imputed_sex[ds.s].isFemale != ds.pheno.isFemale)
+
+    Notes
+    -----
+
+    We have used the same implementation as `PLINK v1.7
+    <http://pngu.mgh.harvard.edu/~purcell/plink/summary.shtml#sexcheck>`__.
+
+    Let `gr` be the the genome reference of the type of the `locus` key (as
+    given by :meth:`.TLocus.reference_genome`)
+
+    1. Filter the dataset to loci on the X contig defined by `gr`.
+
+    2. Calculate alternate allele frequency (AAF) for each row from the dataset.
+
+    3. Filter to variants with AAF above `aaf_threshold`.
+
+    4. Remove loci in the pseudoautosomal region, as defined by `gr`, if and
+       only if `include_par` is ``True`` (it defaults to ``False``)
+
+    5. For each row and column with a non-missing genotype call, :math:`E`, the
+       expected number of homozygotes (from population AAF), is computed as
+       :math:`1.0 - (2.0*maf*(1.0-maf))`.
+
+    6. For each row and column with a non-missing genotype call, :math:`O`, the
+       observed number of homozygotes, is computed interpreting ``0`` as
+       heterozygote and ``1`` as homozygote`
+
+    7. For each row and column with a non-missing genotype call, :math:`N` is
+       incremented by 1
+
+    8. For each column, :math:`E`, :math:`O`, and :math:`N` are combined across
+       variants
+
+    9. For each column, :math:`F` is calculated by :math:`(O - E) / (N - E)`
+
+    10. A sex is assigned to each sample with the following criteria:
+        - Female when ``F < 0.2``
+        - Male when ``F > 0.8``
+        Use `female_threshold` and `male_threshold` to change this behavior.
+
+    **Annotations**
+
+    The returned column-key indexed :class:`.Table` has the following fields:
+
+    - **s** -- The column key of `call`.
```

Comment:
I'm not sure how to fix this, but `s` is not always the name of the field. I think we should just delete `**s** --` and it should just be `The column key(s) of call`.